src/Calculator/StylePanel.xaml.cpp

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "StylePanel.xaml.h"
+
+using namespace CalculatorApp;
+
+using namespace Platform;
+using namespace Platform::Collections;
+using namespace Windows::Foundation;
+using namespace Windows::Foundation::Collections;
+using namespace Windows::UI;
+using namespace Windows::UI::Xaml;
+using namespace Windows::UI::Xaml::Controls;
+using namespace Windows::UI::Xaml::Controls::Primitives;
+using namespace Windows::UI::Xaml::Data;
+using namespace Windows::UI::Xaml::Input;
+using namespace Windows::UI::Xaml::Media;
+using namespace Windows::UI::Xaml::Navigation;
+using namespace Windows::UI::Xaml::Shapes;
+
+DEPENDENCY_PROPERTY_INITIALIZATION(StylePanel, SelectedColor);
+
+StylePanel::StylePanel()
+{
+    InitializeComponent();
+}
+
+void StylePanel::SelectionChanged(Object ^ /*sender */, SelectionChangedEventArgs ^ e)
+{
+    if (e->AddedItems->Size > 0)
+    {
+        SelectedColor = static_cast<SolidColorBrush ^>(e->AddedItems->GetAt(0));
+    }
+}
+
+void StylePanel::OnSelectedColorPropertyChanged(SolidColorBrush ^ /*oldColor*/, SolidColorBrush ^ newColor)
+{
+    SelectColor(newColor);
+}
+
+void StylePanel::ColorChooserLoaded(Object ^ sender, RoutedEventArgs ^ e)
+{
+    SelectColor(SelectedColor);
+}
+
+void StylePanel::SelectColor(SolidColorBrush ^ selectedColor)
```

Comment:
You can also let GridView manage the selection indicator if you modify `GridView::ItemContainerStyle`.

Once done, the only thing you will need to do here will be (in c#, sorry!):   `ColorChooser.SelectedItem = ColorChooser.Items.Where(t => selectedColor.Color.Equals(((SolidColorBrush)t).Color)).FirstOrDefault();` 


```xml
        <Style TargetType="GridViewItem" x:Key="GridViewItemExpanded">
            <Setter Property="Background" Value="Transparent" />
            <Setter Property="TabNavigation" Value="Local" />
            <Setter Property="IsHoldingEnabled" Value="True" />
            <Setter Property="HorizontalContentAlignment" Value="Center" />
            <Setter Property="VerticalContentAlignment" Value="Center" />
            <Setter Property="Margin" Value="0,0,4,4" />
            <Setter Property="AllowDrop" Value="False" />
            <Setter Property="UseSystemFocusVisuals" Value="{StaticResource UseSystemFocusVisuals}" />
            <Setter Property="FocusVisualMargin" Value="-2" />
            <Setter Property="Template">
                <Setter.Value>
                    <ControlTemplate TargetType="GridViewItem">
                        <Grid x:Name="ContentBorder"
                        Control.IsTemplateFocusTarget="True"
                        FocusVisualMargin="{TemplateBinding FocusVisualMargin}">
                            <VisualStateManager.VisualStateGroups>
                                <VisualStateGroup x:Name="FocusStates">
                                    <VisualState x:Name="Focused">
                                        <VisualState.Setters>
                                            <Setter Target="BorderRectangle.Visibility" Value="Collapsed" />
                                        </VisualState.Setters>
                                    </VisualState>
                                    <VisualState x:Name="Unfocused" />
                                </VisualStateGroup>
                                <VisualStateGroup x:Name="CommonStates">
                                    <VisualState x:Name="Normal">
                                        <Storyboard>
                                            <PointerUpThemeAnimation Storyboard.TargetName="Ellipse" />
                                        </Storyboard>
                                    </VisualState>
                                    <VisualState x:Name="PointerOver">
                                        <VisualState.Setters>
                                            <Setter Target="BorderRectangle.Opacity" Value="1"/>
                                            <Setter Target="BorderRectangle.Stroke" Value="{ThemeResource SystemControlHighlightListLowBrush}" />
                                            <Setter Target="ContentBorder.FocusVisualSecondaryBrush" Value="{ThemeResource SystemControlHighlightListLowBrush}" />
                                            <Setter Target="ContentBorder.FocusVisualSecondaryThickness" Value="2" />
                                        </VisualState.Setters>
                                        <Storyboard>
                                            <PointerUpThemeAnimation Storyboard.TargetName="Ellipse" />
                                        </Storyboard>
                                    </VisualState>
                                    <VisualState x:Name="Pressed">
                                        <VisualState.Setters>
                                            <Setter Target="BorderRectangle.Opacity" Value="1"/>
                                            <Setter Target="BorderRectangle.Stroke" Value="{ThemeResource SystemControlHighlightListMediumBrush}" />
                                            <Setter Target="ContentBorder.FocusVisualSecondaryBrush" Value="{ThemeResource SystemControlHighlightListMediumBrush}" />
                                            <Setter Target="ContentBorder.FocusVisualSecondaryThickness" Value="2" />
                                        </VisualState.Setters>
                                        <Storyboard>
                                            <PointerDownThemeAnimation TargetName="Ellipse" />
                                        </Storyboard>
                                    </VisualState>
                                    <VisualState x:Name="Selected">
                                        <VisualState.Setters>
                                            <Setter Target="Ellipse.StrokeThickness" Value="3" />
                                            <Setter Target="ContentBorder.FocusVisualSecondaryBrush" Value="{ThemeResource SystemControlHighlightAccentBrush}" />
                                            <Setter Target="ContentBorder.FocusVisualSecondaryThickness" Value="2" />
                                        </VisualState.Setters>
                                        <Storyboard>
                                            <PointerUpThemeAnimation Storyboard.TargetName="Ellipse" />
                                        </Storyboard>
                                    </VisualState>
                                    <VisualState x:Name="PointerOverSelected">
                                        <VisualState.Setters>
                                            <Setter Target="Ellipse.StrokeThickness" Value="3" />
                                            <Setter Target="ContentBorder.FocusVisualSecondaryBrush" Value="{ThemeResource SystemControlHighlightListAccentMediumBrush}" />
                                            <Setter Target="ContentBorder.FocusVisualSecondaryThickness" Value="2" />
                                        </VisualState.Setters>
                                        <Storyboard>
                                            <PointerUpThemeAnimation Storyboard.TargetName="Ellipse" />
                                        </Storyboard>
                                    </VisualState>
                                    <VisualState x:Name="PressedSelected">
                                        <VisualState.Setters>
                                            <Setter Target="BorderRectangle.Opacity" Value="1"/>
                                            <Setter Target="BorderRectangle.Stroke" Value="{ThemeResource SystemControlHighlightListAccentHighBrush}" />
                                            <Setter Target="ContentBorder.FocusVisualSecondaryBrush" Value="{ThemeResource SystemControlHighlightListAccentHighBrush}" />
                                            <Setter Target="ContentBorder.FocusVisualSecondaryThickness" Value="2" />
                                        </VisualState.Setters>
                                        <Storyboard>
                                            <PointerDownThemeAnimation TargetName="Ellipse" />
                                        </Storyboard>
                                    </VisualState>
                                </VisualStateGroup>
                                <VisualStateGroup x:Name="DisabledStates">
                                    <VisualState x:Name="Enabled" />
                                    <VisualState x:Name="Disabled">
                                        <VisualState.Setters>
                                            <Setter Target="BorderRectangle.Opacity" Value="1"/>
                                        </VisualState.Setters>
                                    </VisualState>
                                </VisualStateGroup>
                            </VisualStateManager.VisualStateGroups>
                            <Ellipse x:Name="Ellipse"
                            HorizontalAlignment="{TemplateBinding HorizontalContentAlignment}"
                            VerticalAlignment="{TemplateBinding VerticalContentAlignment}"
                                     Fill="{TemplateBinding Content}" Height="32" Width="32"
                                     StrokeThickness="0" Stroke="{ThemeResource StylePanelSelectedBrush}"
                            Margin="{TemplateBinding Padding}" />
                            <Rectangle x:Name="BorderRectangle"
                            IsHitTestVisible="False"
                            Stroke="{ThemeResource SystemControlHighlightListAccentLowBrush}"
                            StrokeThickness="2"
                            Opacity="0" />
                        </Grid>
                    </ControlTemplate>
                </Setter.Value>
            </Setter>
        </Style>
```